EosPayload/lib/orcheostrator/orcheostrator.py

```diff
@@ -84,64 +97,125 @@ def valid_driver(driver) -> bool:
+            except KeyError:
+                raise ValueError("Invalid Device ID.")
+
+            used_ids.append(driver_device_id)
+
+            new_driver_class = driver_config['driver_class']
+            if new_driver_class in available_drivers:
+                for field in available_drivers.get(new_driver_class).get_required_device_config():
+                    if driver_config.get("driver_settings") is None or driver_config.get("driver_settings").get(field) is None:
+                        raise ValueError("Required driver setting not provided.")
+                enabled_driver_list.append((available_drivers[new_driver_class], driver_config))
+            else:
+                raise ValueError("Driver class does not exist.")
+
+        config = {"enabled_drivers_list":enabled_driver_list}
+        return config
+
     #
     # PROTECTED HELPER METHODS
     #
 
     def _spawn_drivers(self) -> None:
         """ Enumerates over all the classes in the drivers dir and spins them up """
         self._logger.info("Spawning Drivers")
-        for attribute_name in dir(drivers):
-            driver = getattr(drivers, attribute_name)
-            if self.valid_driver(driver):
-                container = DriverContainer(driver)
-                try:
-                    if driver.get_device_id() is None:
-                        self._logger.error(f"can't spawn process for device from class '{driver.__name__}'"
-                                           " because device id is not defined")
-                        container.update_status(Status.INVALID)
-                        self._drivers['<' + driver.__name__ + '>'] = container
-                        continue
-                    if not driver.enabled():
-                        self._logger.warning(f"skipping device '{driver.get_device_pretty_id()}' from"
-                                             f" class '{driver.__name__}' because it is not enabled")
-                        container.update_status(Status.DISABLED)
-                        self._drivers[driver.get_device_id()] = container
-                        continue
-                    self._logger.info(f"spawning process for device '{driver.get_device_pretty_id()}' from"
-                                      f" class '{driver.__name__}'")
-                    proc = Process(target=self._driver_runner, args=(driver, self.output_directory), daemon=True)
-                    container.process = proc
-                    container.update_status(Status.HEALTHY, 1)
-                    proc.start()
-                    self._drivers[driver.get_device_id()] = container
-                except Exception as e:
-                    self._logger.critical("A fatal exception occurred when attempting to load driver from"
-                                          f" class '{driver.__name__}': {e}\n{traceback.format_exc()}")
+        driver_list = self.config["enabled_drivers_list"]
```

Comment:
a major side effect here is that orcheostrator health check has no visibility as to what drivers are disabled

Reply:
Do you think the health check needs that visibility, or would just logging disabled/unconfigured drivers be adequate?

Reply:
logging would probably suffice yeah, in which case the DISABLED state should be removed

Reply:
Done